examples/read_write/read_write.c

```diff
@@ -53,7 +52,7 @@ int main(int argc, char *argv[]) {
 	} else if ((strcmp(argv[2], "write") == 0) && (argc == 5)) {
 		g_operation = WRITE;
 
-		if ((strlen(argv[4]) >= 2) && (argv[4][0] == '0') && (argv[4][0] == 'x')) {
+		if ((strlen(argv[4]) >= 2) && (argv[4][0] == '0') && ((argv[4][1] == 'x') || (argv[4][1] == 'X'))) {
```

Comment:
Thanks for finding this issue. But would actually in case of hexadecimal we should reverse the condition such as:

```
if ((strlen(argv[4]) >= 2) && (argv[4][0] == '0') && ((argv[4][1] == 'x') || (argv[4][1] == 'X'))) {
	value_data = strtol(argv[4], NULL, 16);
} else {
	value_data = strtol(argv[4], NULL, 0);
}
```